code/game/machinery/recharger.dm

```diff
@@ -35,30 +35,27 @@
 	..()
 
 /obj/machinery/recharger/attackby(obj/item/weapon/G, mob/user)
-	if(istype(user,/mob/living/silicon))
-		return
+	if(issilicon(user))
+		return 1
+	if(stat & (NOPOWER | BROKEN))
+		to_chat(user, "<span class='notice'>[src] isn't connected to a power source.</span>")
+		return 1
+	if(charging)
+		to_chat(user, "<span class='warning'>There's \a [charging] already charging inside!</span>")
+		return 1
+	if(!anchored)
+		to_chat(user, "<span class='warning'>You must secure \the [src] before you can make use of it!</span>")
+		return 1
+	if(istype(G, /obj/item/weapon && !user.drop_item(G, src)))
```

Comment:
Doesn't this drop the item into the machine whether or not its the correct item type